src/client/QXmppRosterManager.cpp

```diff
@@ -613,7 +613,16 @@ void QXmppRosterManager::onRegistered(QXmppClient *client)
 
         auto exportData = [this]() {
             return chainMapSuccess(requestRoster(), this, [](QXmppRosterIq &&iq) -> RosterData {
-                return { iq.items() };
+                auto items = iq.items();
+
+                // We don't want this to be sent while importing.
+                // See https://datatracker.ietf.org/doc/html/rfc6121#section-2.1.2.2
+                for (int i = 0; i < items.count(); ++i) {
+                    auto &item = items[i];
+                    item.setSubscriptionStatus({});
+                }
```

Comment:
please use range-based for loop

Reply:
That would detach the Qt container

Reply:
`items[i]` also detaches the Qt container. We're modifying the values here, so we must detach the container, right?